gossip/components/processor.c

```diff
@@ -7,269 +7,384 @@
 
 #include <string.h>
 
-#include "ciri/core.h"
-#include "common/storage/sql/defs.h"
 #include "gossip/components/processor.h"
+#include "gossip/node.h"
 #include "utils/containers/lists/concurrent_list_neighbor.h"
-#include "utils/containers/queues/concurrent_queue_packet.h"
 #include "utils/logger_helper.h"
 
-#define PROCESSOR_COMPONENT_LOGGER_ID "processor_component"
+#define PROCESSOR_LOGGER_ID "processor"
+#define PROCESSOR_TIMEOUT_SEC 5
 
-static retcode_t process_transaction_bytes(processor_state_t *const state,
+/*
+ * Private functions
+ */
+
+/**
+ * Converts transaction bytes from a packet to a transaction, validates it and
+ * checks its solidity.
+ * If valid and new: stores and broadcasts it.
+ *
+ * @param processor The processor state
+ * @param neighbor The neighbor that sent the packet
+ * @param packet The packet from which to process transaction bytes
+ * @param hash A hash to be filled with transaction bytes hash
+ *
+ * @return a status code
+ */
+static retcode_t process_transaction_bytes(processor_t const *const processor,
                                            neighbor_t *const neighbor,
-                                           iota_packet_t *const packet,
-                                           iota_transaction_t *const tx) {
+                                           iota_packet_t const *const packet,
+                                           flex_trit_t *const hash) {
   retcode_t ret = RC_OK;
-  bool exists = false;
-
-  if (state == NULL) {
-    return RC_PROCESSOR_COMPONENT_NULL_STATE;
-  } else if (neighbor == NULL) {
-    return RC_PROCESSOR_COMPONENT_NULL_NEIGHBOR;
-  } else if (packet == NULL) {
-    return RC_PROCESSOR_COMPONENT_NULL_PACKET;
-  } else if (tx == NULL) {
-    return RC_PROCESSOR_COMPONENT_NULL_TX;
+
+  if (processor == NULL || neighbor == NULL || packet == NULL || hash == NULL) {
+    return RC_NULL_PARAM;
   }
 
-  if (true /* TODO(thibault): if !cached */) {
-    flex_trit_t tx_flex_trits[FLEX_TRIT_SIZE_8019];
-
-    flex_trits_from_bytes(tx_flex_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
-                          packet->content, NUM_TRITS_SERIALIZED_TRANSACTION,
-                          NUM_TRITS_SERIALIZED_TRANSACTION);
-    if ((*tx = transaction_deserialize(tx_flex_trits)) == NULL) {
-      neighbor->nbr_invalid_tx++;
-      log_warning(PROCESSOR_COMPONENT_LOGGER_ID,
-                  "Deserializing transaction failed\n");
-      return RC_PROCESSOR_COMPONENT_INVALID_TX;
+  // TODO Check if transaction hash is cached
+
+  if (false) {
+  } else {
+    bool exists = false;
+    struct _iota_transaction transaction = {.snapshot_index = 0, .solid = 0};
+    flex_trit_t transaction_flex_trits[FLEX_TRIT_SIZE_8019];
+    trit_array_t key = {.trits = hash,
```

Comment:
As far as I can tell, you're never computing the hash and thus never writing any values to `hash`

Reply:
Nevermind, below is a `memcpy` :sweat_smile: